sourcegen-model/src/main/java/io/micronaut/sourcegen/model/ExpressionDef.java

```diff
@@ -481,6 +494,19 @@ record NewInstance(ClassTypeDef type,
                        List<ExpressionDef> values) implements ExpressionDef {
     }
 
+    /**
+     * The new instance expression.
+     *
+     * @param type   The type
+     * @param value The initial value
+     * @author Denis Stepanov
+     * @since 1.0
+     */
+    @Experimental
+    record PrimitiveInstance(TypeDef type,
```

Comment:
Please move this to `TypeDef.Primitive`, so we call `TypeDef.Primitive.INT.instantiate(1)`